pyigm/field/igmfield.py

```diff
@@ -0,0 +1,236 @@
+""" Class for correlating galaxies and IGM within a given field on the sky
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import warnings
+import pdb
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy import constants as const
+
+#from xastropy.xutils import xdebug as xdb
+
+
+class IgmGalaxyField(object):
+    """ Class for a field associating galaxies to the IGM/CGM
+
+    Parameters
+    ----------
+    radec : tuple or SkyCoord
+        (RA,DEC) in deg or astropy.coordinate
+    name : str, optional
+    """
+
+    # Initialize 
+    def __init__(self, radec, name='', cosmo=None, verbose=False):
+        # coord
+        if isinstance(radec, (tuple)):
+            self.coord = SkyCoord(ra=radec[0], dec=radec[1])
+        elif isinstance(radec, SkyCoord):
+            self.coord = radec
+
+        # Field
+        self.name = name
```

Comment:
I suppose this gets back to our discussion last week where we agreed that a 'field' is a observational construct, but I suggest that procedurally we do not impose such a naming convention.  Because a 'field' is an observational construct, it often exists centered on a bright QSO, which may have a historically prevalent name (and may not even refer to J2000 coordinates as in 'PG1407+265') that is unambiguous in communication formally and informally.  The naming convention is certainly okay if another name is not provided!  We just shouldn't impose its adherence in practice.